File: abc/405/py/b.py
def sv3(M: int, A: list[int]) -> int:
    visited = [True] + [False] * M
    unique_count = 0
    for i, e in enumerate(A):
        if not visited[e]:
            visited[e] = True
            unique_count += 1
        if unique_count == M:
            return len(A) - i
    return 0

def sv2(M: int, A: list[int]) -> int:
    set1 = set()
    for i, e in enumerate(A):
        set1.add(e)
        if len(set1) == M:
            return len(A) - i
    return 0

def sv1(M: int, A: list[int]) -> int:
    B = [1] + [0] * M 
    for i, e in enumerate(A):
        B[e] += 1
        if B.count(0) == 0:
            return len(A) - i
    return 0

def main():
    N, M = map(int, input().split())
    A = list(map(int, input().split()))
    res = sv3(M, A)
    print(res)

if  __name__ == '__main__':
    main()

File: abc/405/c/b.c
#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>

// sv2 funksiyasining C tilidagi varianti
int sv2(int M, int N, int A[]) {
    // Elementlarni tekshirib turish uchun dinamik "visited" massivi
    // Maksimal element qiymatini topish yoki masaladagi M ga qarab o'lcham olish kerak.
    // Biz bu yerda elementlar qiymati M dan oshmaydi deb hisoblaymiz (Pythondagi sv1 mantiqiga o'xshash)
    bool* visited = (bool*)calloc(M + 1, sizeof(bool));
    
    int unique_count = 0;

    for (int i = 0; i < N; i++) {
        int element = A[i];
        
        // Agar bu elementni birinchi marta ko'rayotgan bo'lsak
        if (!visited[element]) {
            visited[element] = true;
            unique_count++; // Unikal elementlar sonini oshiramiz
        }
        
        // Agar unikal elementlar soni M ga yetsa
        if (unique_count == M) {
            free(visited); // Xotirani bo'shatamiz
            return N - i;
        }
    }

    free(visited); // Xotirani bo'shatamiz
    return 0;
}

int main() {
    int N, M;
    
    // N va M ni o'qib olamiz
    if (scanf("%d %d", &N, &M) != 2) return 0;

    // Dinamik massiv yaratamiz
    int* A = (int*)malloc(N * sizeof(int));
    for (int i = 0; i < N; i++) {
        scanf("%d", &A[i]);
    }

    // Natijani hisoblaymiz
    int res = sv2(M, N, A);
    printf("%d\n", res);

    // Ajratilgan xotiralarni tozalaymiz
    free(A);

    return 0;
}